### PR TITLE
Add DataFrame.plot.kde() (an alias of 'density')

### PR DESCRIPTION
**Pull request by itholic**
This PR implements kde in DataFrame. It reuses Series' implementation. at #767  

like Series's kde plot, Since DataFrame's also uses MLlib's KernelDensity API to calculate KDE , so slightly different from pandas but seems good enough either:

스크린샷 2019-09-17 오후 2 24 34
스크린샷 2019-09-17 오후 2 25 18

And also kde is an alias of 'density', you can get exactly same result when you use 'density' rather than 'kde' like below:

스크린샷 2019-09-17 오후 2 27 30
스크린샷 2019-09-17 오후 2 27 09

**Multiple columns examples:**

스크린샷 2019-09-17 오후 2 55 35

and for each row as Series.plot.kde looks same like below:

스크린샷 2019-09-17 오후 2 57 57

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
@itholic, can you show an example with more than two columns?

**Comment by itholic** [comment]
@HyukjinKwon Sure, Here is
스크린샷 2019-09-17 오후 2 55 35

and for each row as Series.plot.kde looks same like below:

스크린샷 2019-09-17 오후 2 57 57

**Comment by HyukjinKwon** [comment]
@itholic, looks good. Can you fix the tests?

**Comment by itholic** [comment]
@HyukjinKwon i'm still working on it, but these tests are still failed, i'm trying to find out solution.

**Comment by ueshin** [comment]
@HyukjinKwon Let me share the question here, too. https://github.com/databricks/koalas/pull/767#discussion_r328350702 which might relate to this PR.

**Comment by HyukjinKwon** [comment]
@itholic can you check ^? I'll check it tomorrow.

**Comment by itholic** [comment]
@HyukjinKwon Sure, i'll check this out. sorry to late reply (it was very busy today 😭 )

File: databricks/koalas/plot.py [hunk 1]
```diff
@@ -631,7 +631,6 @@ def _plot(
             cls, ax, y, style=None, bw_method=None, ind=None,
             column_num=None, stacking_id=None, **kwds):
         # 'y' is a Spark DataFrame that selects one column.
-
```
Comment (HyukjinKwon) [inline databricks/koalas/plot.py]: Revert unrelated changes back.

File: databricks/koalas/plot.py [hunk 2]
```diff
@@ -919,7 +918,6 @@ def _plot(data, x=None, y=None, subplots=False,
     if kind in ('scatter', 'hexbin'):
         plot_obj = klass(data, x, y, subplots=subplots, ax=ax, kind=kind, **kwds)
     else:
-
```
Comment (HyukjinKwon) [inline databricks/koalas/plot.py]: this one too

**Comment by HyukjinKwon** [comment]
There seems an issue about plot comparison. plot image has to be closed for each plot but seems not. So, overlapped plot images are being compared. For instance,

we should compare:

a
b

but now we're comparing:

a1
b1

File: databricks/koalas/tests/test_frame_plot.py [hunk 3]
```diff
@@ -335,10 +335,26 @@ def _test_hist_plot(pdf, kdf):
...
+        ax2 = kdf.plot.kde(bw_method=3)
+        self.compare_plots(ax1, ax2)
+
+        ax1 = pdf.plot(kind='kde', bw_method=3)
+        ax2 = kdf.plot(kind='kde', bw_method=3)
+        self.compare_plots(ax1, ax2)
+
+        ax1 = pdf.plot.kde(bw_method=3, ind=[1, 2, 3])
+        ax2 = kdf.plot.kde(bw_method=3, ind=[1, 2, 3])
+        self.compare_plots(ax1, ax2)
```
Comment (HyukjinKwon) [inline databricks/koalas/tests/test_frame_plot.py]: Can you comment out this test @itholic with the explanation above? Let's merge this one first.

Comment (HyukjinKwon) [reply databricks/koalas/tests/test_frame_plot.py]: We should fix whole test cases except only some test cases in plot - it seems it was wrong at the very first place.

**Comment by itholic** [comment]
@HyukjinKwon Thanks for review !! i just resolved all about you commented.

**Comment by HyukjinKwon** [comment]
@ueshin, I am going to rewrite the plot tests soon. Let me merge this for now.

**Comment by ueshin** [comment]
Sounds good, thanks!

**Comment by HyukjinKwon** [comment]
I made a PR https://github.com/databricks/koalas/pull/838

BTW, @itholic, can you add some examples for plot implementations in their docstrings? Basically we finished the rough implementation of all plots (except 2 in DataFrame). So it should be better to have some examples.

Can you also check if we're able to post some images? e.g. https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.plot.box.html#pandas.DataFrame.plot.box

**Comment by itholic** [comment]
@HyukjinKwon sure, I'm going to add & check them 😃

File: docs/source/reference/frame.rst [hunk 4]
```diff
@@ -243,4 +243,6 @@ specific plotting methods of the form ``DataFrame.plot.<kind>``.
    DataFrame.plot.line
    DataFrame.plot.pie
    DataFrame.plot.scatter
-   DataFrame.hist
+   DataFrame.plot.hist
```
Comment (ueshin) [inline docs/source/reference/frame.rst]: Seems like `DataFrame.plot.hist` is listed twice.
I'm not sure we should keep `DataFrame.hist`, though.

Comment (HyukjinKwon) [reply docs/source/reference/frame.rst]: @itholic can you fix this? `DataFrame.hist` seems in the pandas doc as an alias. Let's keep it for now.

Comment (itholic) [reply docs/source/reference/frame.rst]: sure, fixed in #851 
Thanks for comment !!
